aikau/src/main/resources/alfresco/renderers/Thumbnail.js

```diff
@@ -408,31 +512,40 @@ define(["dojo/_base/declare",
       onLinkClick: function alfresco_renderers_Thumbnail__onLinkClick(evt) {
          event.stop(evt);
 
-         // TODO: Need to use a nodeRef property attribute that can be configured
-         var nodeRef = lang.getObject("nodeRef", false, this.currentItem);
-         if (!nodeRef)
-         {
-            nodeRef = lang.getObject("node.nodeRef", false, this.currentItem);
-         }
+         // Delegate to the ItemSelectionMixin - this will check if select on click is supported...
+         this.onSelectionClick();
 
-         // Check to see if the thumbnail is configured to display previews when clicked,
-         // this particular type of action could require an XHR request of the full node
-         // data so it needs to be handled in a specific way...
-         if (this.showDocumentPreview)
+         // Check whether or not the thumbnail should ONLY support selection on click (because in 
+         // some circumstances, e.g. film strip view carousel we might want to select AND perform
+         // an additinal action)...
+         if (!this.onlySelectOnClick)
          {
-            this.nodePromise = this.currentItem;
-            var type = lang.getObject("node.type", false, this.currentItem),
-                mimetype = lang.getObject("node.mimetype", false, this.currentItem);
-            if (!type || !mimetype)
+            // TODO: Need to use a nodeRef property attribute that can be configured
```

Comment:
No, it means that we could be using something like itemKeyProperty, but with the inconsistent REST APIs this is quite tricky to do. For backwards compatibility we'd need to check the itemKeyProperty and then "nodeRef" and "node.nodeRef"... it would all get a bit messy and is also out of scope for this issue.